onedal/datatypes/_data_conversion.py

```diff
@@ -15,9 +15,17 @@
 # ===============================================================================
 
 import warnings
+from onedal import _is_dpc_backend
 from onedal import _backend
 from daal4py.sklearn._utils import make2d
 
+try:
+    import dpctl
+    import dpctl.tensor as dpt
+    dpctl_available = dpctl.__version__ >= '0.14'
+except ImportError:
+    dpctl_available = False
```

Comment:
This code is duplicated multiple times across repo. Move it to dedicated file

Reply:
Good point. Let me implement it as a feature in other PR.

Reply:
There is no problem to implement it here

Reply:
These kind of refactoring is not a purpose of this PR. It should be logically in other  PR. As you mentioned it's duplicated several times, we need some common primitives for that.